protocol/__init__.py

```diff
@@ -0,0 +1,136 @@
+        log.info('w3 provider set')
+        return w3
+
+    def initialize_datatrust(self):
+        """
+        Confirm or create role as datatrust backend in protocol
+        """
+        log.info('Getting current datatrust address from network')
+        self.datatrust = Datatrust(self.datatrust_wallet)
+        self.datatrust.at(self.w3, self.datatrust_contract)
+
+        backend = call(self.datatrust.get_backend_address())
+        if backend == self.datatrust_wallet:
+            log.info('This server is the datatrust host. Resolving registration')
+            resolve = send(
+                self.w3, 
+                self.datatrust_key, 
+                self.datatrust.resolve_registration(
+                    self.w3.sha3(text=self.datatrust_host), {'from': self.datatrust_wallet}
+                    )
+                )
+            log.info(f'Resolved, transaction id: {resolve}')
+        else:
+            # backend not set, or is set to a different host
+            datatrust_url = call(self.datatrust.get_backend_url())
+            if datatrust_url == self.datatrust_host:
+                log.info('Server has been registered as datatrust, but not voted in')
+            else:
+                log.info('No backend or different host set. Resolving prior registrations and Submitting this one for voting')
+                datatrust_hash = self.w3.sha3(text=self.datatrust_host)
+                #TODO: Get the hash of any pending candidates and resolve before registration
+                resolve = send(
+                    self.w3, 
+                    self.datatrust_key, 
+                    self.datatrust.resolve_registration(
+                        self.w3.sha3(text=self.datatrust_host), {'from': self.datatrust_wallet}
+                        )
+                    )
+                log.info(f'Resolved any prior registration, transaction id: {resolve}')
+                register = self.register_host()
+                log.info(f'Datatrust has been registered, transaction id: {register}')
+
+    def wait_for_vote(self):
+        """
+        Check if this backend is registered as the datatrust
+        """
+        is_host = False
+        while is_host == False:
+            backend = call(self.datatrust.get_backend_url())
+            if backend != self.datatrust_host:
+                print('backend not voted in yet, waiting for votes...')
+                sleep(30)
+            else:
+                is_host = True
+        return True
+
+    def register_host(self):
+        """
+        Register a host as the datatrust in protocol
+        """
+        log.info('****Registering host****')
+        register = send(self.w3, self.datatrust_key, self.datatrust.register(self.datatrust_host, {'from': self.datatrust_wallet}))
+        voting = Voting(self.datatrust_wallet)
+        voting.at(self.w3, self.voting_contract)
+        datatrust_hash = self.w3.sha3(text=self.datatrust_host)
```

Comment:
i think they are deprecating that sha3 in favor of `w3...keccak`

Reply:
I found the EIP where it was agreed upon and merged, but I can't find where it's been implemented in web3.js or web3.py. Do you know where it's at in web3.py?

Reply:
I don't see it in the 4.9.2 docs yet so fuck em. leave it `sha` til they decide to write it down